### ISSUE TITLE
HL7 Send Error

### ISSUE THREAD

**Issue by revren** [RobHolme/vscode-hl7tools#12]
When sending a HL7 message the destination system seems to have an issue with files originating from VSCode. The very same file works fine in the destination system.
Is there something the extension might be doing with these files with line endings or something?

**Comment by Lakkanna** [comment]
Every line is ending with space is that making any problem

**Comment by revren** [comment]
> Every line is ending with space is that making any problem
There is no whitespace at the end of the file or each segment

**Comment by revren** [comment]
```
MSH|^~\&|||||20190506131759||ADT^A05|191919|T|2.5
EVN|A05|20190506131759||HE
PID|1||ABC12345^^^^JHN||NAME1^NAME2|MAIDEN|19520918|F|||4 The Street^^1ABC^DEF^KJL^123^^^^R||999999999^^PH
PV1|1|T|||||00000000^MEDNAME^MEDNAME^12|||||||||||||||||||||||||||||||||||||20190506131759

```

^^My HL7 file. When I send that via a HL7 simulator and an integration engine my destination system manages it ok. From VSCode and HL7 Tools ext though I get a parsing error at my destination system.

I suspect some sort of error with either: message encoding, line feeds or perhaps the LLP delimiters?

**Comment by revren** [comment]
Its strange because the destination system sees 277 bytes for the message from VSCode and those from the other systems. I cannot see any difference in the message content in its logs either.

**Comment by revren** [comment]
I am using VSCode Insiders 1.36. 

Retried with VSCode standard 1.33.1 - Same issue

**Comment by revren** [comment]
(Im a noob with extension editing so please be kind).......

I have tried to hard code the text for the LLP delimiters: 
`const BEG = '\x0B'
const END = '\x1C\x0D'`

That didnt work.

I have also tried to enforce utf8 (after adding it to the dependencies):
`client.write(utf8.encode(BEG + HL7Message + END));`

That didnt work either. :-(

**Comment by RobHolme** [comment]
I've published an updated version of the extension that I hope will fix this issue (v1.9.8). 

The extension was failing to send the correct end of segment terminator (x0D) for each segment in the message. Instead the end of line terminator (LF or CRLF) was still left at the end of each segment. It seems most systems I tested with tolerated this, but I assume this would cause issues with systems that are strict about using the correct end of segment terminator.

I haven't had the chance to test this thoroughly - could you let me know if the updated version of the extension corrects the issue for you?

regards,
Rob.

**Comment by revren** [comment]
Hi @RobHolme thank you so much! That has worked. I tested a number of different messages and the destination system accepted them.
